src/pages/manage_dac/ManageDac.js

```diff
@@ -114,6 +119,37 @@ class ManageDac extends Component {
     });
   };
 
+  deleteDac = async (selectedDac) => {
+    await DAC.datasets(selectedDac.dacId).then((response) => {
+      isEmpty(response) ?
+        this.setState(prev => {
+          prev.showConfirmationModal = true;
+          prev.isEditMode = false;
+          prev.selectedDac = selectedDac;
+          return prev;
+        })
+        : Notifications.showError({text: "All datasets assigned to this DAC must be reassigned before this can be deleted."});
+    });
+  };
```

Comment:
We should be able to tell beforehand if the DAC has datasets and therefore the icon shouldn't be enabled and instead show a tooltip that has the error message in it. There are probably a few ways to do this, but I think the simplest way is to make a new API call that gets all DACs along with their Datasets so you don't have to make this extra call per DAC.

Reply:
Even better, we already have a datasetids property on the DAC, but it isn't populated in the query we're making for all DACs. We should update the underlying query so this value is always populated and you would not need to make the extra call for datasets by dacid.

Reply:
I think this will do it:
```
    @SqlQuery("SELECT distinct d.*, a.datasetid " +
            " FROM dac d " +
            " LEFT JOIN consents c ON d.dac_id = c.dac_id " +
            " LEFT JOIN consentassociations a ON a.consentid = c.consentid ")
    List<Dac> findAll();
```

Reply:
good idea!

Reply:
so there is only one datasetid being returned here, because theres only one on a consent association, but a dac could have any number of datasets, so we are missing some information I think but I'm not really sure how to solve this problem

Reply:
We have to be able to query for this, so that's where the work needs to happen. It could be the mapper as well - we might need to change how we're mapping DACs from the result set.